### DISCUSSION TITLE
Authentication for App

### DISCUSSION THREAD

**Discussion by manishonc** [pocketbase/pocketbase#2979] [Q&A]
Sorry to bother you again.

I am working on an educational app that doesn't require any user authentication. All the content is accessible in the app.

But I want to know if there is any solution that I can use to authenticate the app, like a bearer token or something similar. This way, I can authenticate the app while making API requests to the Pocketbase backend. This will ensure that even if anybody gets hold of the API URL, they will not be able to access it because they won't have the authentication key.

Thanks @ganigeorgiev

**Comment by ganigeorgiev** [comment]
I'm not sure I follow, but we use JWT tokens for the app users and admin authentication, if that's what you are asking.
If you haven't already you can also check https://pocketbase.io/docs/authentication/.

**Reply by manishonc** [reply]
The app is open without authentication, meaning no login is required to use the app. I just want authentication on API requests called within the app. This way, the REST API can only be accessed by the app.

**Reply by ganigeorgiev** [reply]
> I just want authentication on API requests called within the app.

If your app is client-side (eg. browser SPA or mobile) then it shouldn't really matter in your case because whatever auth mechanism you choose the user will have access to the token anyway if you hardcode it in the app.

If you are communicating with PocketBase only server-side, then you can "lock" your collections and require admin authentication. We don't have fixed admin tokens or API keys at the moment, if that is what you are looking for (_it is in my todo to consider it but remains a low priority_).

If you want to authenticate as admin in a node server-side application, you can check for examples in https://github.com/pocketbase/pocketbase/discussions/2908#discussioncomment-6435951 and  https://github.com/pocketbase/pocketbase/discussions/2922#discussioncomment-6470104.
